app/controllers/balance_transactions_controller.rb

```diff
@@ -0,0 +1,73 @@
+class BalanceTransactionsController < ApplicationController
+  before_action :authenticate_user!
+  before_action :create_new_form, only: %i[new create]
+  before_action :create_edit_form, only: %i[edit update]
+
+  def new
+  end
+
+  def create
+    if @form.validate(create_params)
+      BalanceTransactions::Creator.new(create_params).call
+      flash[:notice] = t('transaction_create')
+    else
+      flash[:alert] = t('transaction_not_create')
+    end
+
+    respond_to do |format|
+      format.html { redirect_to activity_page_path }
+      format.js
+    end
+  end
+
+  def edit
+  end
+
+  def update
+    if @form.validate(update_params)
+      BalanceTransactions::Updater.new(update_params).call
+      flash[:notice] = t('transaction_update')
+    else
+      flash[:alert] = t('transaction_not_update')
+    end
+
+    respond_to do |format|
+      format.html { redirect_to activity_page_path }
+      format.js
+    end
+  end
+
+  private
+
+  def transaction
+    Transaction.find(params[:id])
```

Comment:
@marderer let's memorize value in order not to make request to db after first call
```ruby
@_transaction || = 
```